hdf/src/hblocks.c

```diff
@@ -1642,11 +1627,11 @@ HLPcloseAID(accrec_t *access_rec)
         /* free the linked list of links/block tables */
         for (t_link = info->link; t_link; t_link = next) {
```

Comment:
info here is set to "access_rec->special_info" which could freed in HLIstaccess failure.